File: fake/Dii_TheaterManagement_Bff.Fake/Data/Booking.cs
using System;
using System.Text.Json.Serialization;
// test
namespace FakeTheaterBff.Data
{
    public class Booking
    {
        //test
        public Booking()
        {

        }
        public Booking(string theaterCode, int monthId, Guid movieId)
        {
            TheaterCode = theaterCode;
            MonthId = monthId;
            MovieId = movieId;
        }

        public Guid MovieId { get; set; }
        public Movie Movie { get; set; }

        /// <summary>
        /// One of two keys that make up <see cref="Booking"/>'s composite key.
        /// </summary>
        public string TheaterCode { get; set; }
        [JsonIgnore] // Do not serialize to the API (but do save to database)
        public DetailedTheater DetailedTheater { get; set; }

        /// <summary>
        /// One of two keys that make up <see cref="Booking"/>'s composite key.
        /// </summary>
        public int MonthId { get; set; }
    }
}

File: src/app/core/singleton-services/bookings/Booking.ts
import { Movie } from "../movies/movie";

export class Booking {
    movieId!: string;
    movie: Movie;
    theaterCode: string;
    monthId!: number;

    constructor(movieId: string, movie: Movie, theaterCode: string, monthId: number){
        this.movieId = movieId;
        this.movie = movie;
        this.theaterCode = theaterCode;
        this.monthId = monthId;
      }
}
